mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -171,17 +171,18 @@ public void delete(Mailbox mailbox, MailboxMessage message) {
     @Override
     public Iterator<MailboxMessage> findInMailbox(Mailbox mailbox, MessageRange messageRange, FetchType ftype, int max) {
         CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
-        return retrieveMessages(retrieveMessageIds(mailboxId, messageRange), ftype, Limit.from(max))
+        return retrieveMessageIds(mailboxId, messageRange)
+            .flatMap(ids -> retrieveMessages(ids, ftype, Limit.from(max)))
```

Comment:
`flatMapSequential` instead of using `sort` may be better

Reply:
ids or not sorted when calling `retrieveMessageIds`